src/components/SellNft/index.tsx

```diff
@@ -1,70 +1,196 @@
-import React from 'react';
-import { useForm, Controller } from 'react-hook-form';
-import { LAMPORTS_PER_SOL } from '@solana/web3.js';
-import { Link } from 'react-router-dom';
+import React, { useState }  from 'react'
+import { useForm, Controller } from 'react-hook-form'
+import { Transaction, PublicKey, LAMPORTS_PER_SOL } from '@solana/web3.js'
+import { Link } from 'react-router-dom'
+import { AuctionHouseProgram } from '@metaplex-foundation/mpl-auction-house'
+import * as anchor from '@project-serum/anchor'
+import { NATIVE_MINT } from '@solana/spl-token'
+import { useConnection, useWallet } from '@solana/wallet-adapter-react'
+
+const { createSellInstruction } = AuctionHouseProgram.instructions
 
 interface OfferForm {
-  amount: string;
+  amount: string
 }
 
 interface OfferProps {
-  nft: any;
+  nft: any
 }
 
 const SellNft = ({ nft }: OfferProps) => {
-  const { control, watch } = useForm<OfferForm>({});
+  const { control, watch, register, handleSubmit } = useForm<OfferForm>({})
+  const { publicKey, signTransaction } = useWallet()
+  const { connection } = useConnection()
+  const [sellAmount, setSellAmount] = useState(0)
+
+  const sellNftTransaction = async () => {
+    const sellPrice = String(Number(sellAmount) * LAMPORTS_PER_SOL)
+
+    // NOTE: always 1 for NFTs since they are spl token with count of 1
+    const tokenSize = '1'
+    // TODO: read auction house from indexer
+    const auctionHouse = new PublicKey('BXUEBP3meoskWuU6aksqKKAcvRNLPzhjqL5RFeUd5vAR')
+    console.log(auctionHouse)
+    // TODO: read authority from auction house record (auctionHouse.authority)
+    const authority = new PublicKey(
+      'BXUEBP3meoskWuU6aksqKKAcvRNLPzhjqL5RFeUd5vAR'
+    )
+    const auctionHouseFeeAccount = new PublicKey(
+      'BXUEBP3meoskWuU6aksqKKAcvRNLPzhjqL5RFeUd5vAR'
+    )
+    // TODO: compute ata for owner with token_account and metadata.mint
+    const tokenAccount = new PublicKey(
+      'BXUEBP3meoskWuU6aksqKKAcvRNLPzhjqL5RFeUd5vAR'
+    )
+    const tokenMint = new PublicKey(nft.mintAddress)
+
+    if (!publicKey || !signTransaction) {
+      return
+    }
+
+    // Find TradeState Account
+    const [
+      sellerTradeState,
+      tradeStateBump,
+    ] = await AuctionHouseProgram.findTradeStateAccount(
+      publicKey,
+      auctionHouse,
+      tokenAccount,
+      NATIVE_MINT,
+      tokenMint,
+      sellPrice,
+      tokenSize
+    )
+
+    const associatedTokenAccount = (
+      await AuctionHouseProgram.getAtaForMint(tokenMint, new PublicKey(nft.owner.address)) 
+    )[0]
+
+    const metadata = await AuctionHouseProgram.getMetadata(tokenMint)
+
+    const [
+      programAsSigner,
+      programAsSignerBump,
+    ] = await AuctionHouseProgram.getAuctionHouseProgramAsSigner()
```

Comment:
I changed these to find<>Address to align with the underlying use of findAddress